plugins/inputs/sysstat/sysstat.go

```diff
@@ -178,7 +179,12 @@ func (s *Sysstat) collect() error {
 	for _, act := range s.Activities {
 		options = append(options, "-S", act)
 	}
-	s.tmpFile = path.Join("/tmp", fmt.Sprintf("sysstat-%d", time.Now().Unix()))
+
+	var err error
+	s.tmpFile, err = createTempFile()
```

Comment:
Look into having this either not be a struct field, or perhaps leave it on the struct and reuse a directory each interval, removing only when the plugin is stopped.